runners/flink/src/test/java/org/apache/beam/runners/flink/FlinkExecutionEnvironmentsTest.java

```diff
@@ -438,8 +424,34 @@ public void shouldSetSavepointRestoreForRemoteStreaming() {
             options, Collections.emptyList());
     // subject to change with https://issues.apache.org/jira/browse/FLINK-11048
     assertThat(sev, instanceOf(RemoteStreamEnvironment.class));
-    assertThat(
-        Whitebox.getInternalState(sev, "restoreSettings"),
-        is(SavepointRestoreSettings.forPath(path)));
+    assertThat(getSavepointPath(sev), is(path));
+  }
+
+  private void checkHostAndPort(Object env, String expectedHost, int expectedPort) {
+    try {
+      assertThat(Whitebox.getInternalState(env, "host"), is(expectedHost));
+      assertThat(Whitebox.getInternalState(env, "port"), is(expectedPort));
+    } catch (Throwable t) {
+      // for flink 1.10+
+      String host =
+          ((Configuration) Whitebox.getInternalState(env, "configuration"))
+              .getString(RestOptions.ADDRESS);
+      int port =
+          ((Configuration) Whitebox.getInternalState(env, "configuration"))
+              .getInteger(RestOptions.PORT);
+      assertThat(
+          new InetSocketAddress(host, port), is(new InetSocketAddress(expectedHost, expectedPort)));
+    }
+  }
+
+  private String getSavepointPath(Object env) {
+    try {
+      return ((SavepointRestoreSettings) Whitebox.getInternalState(env, "restoreSettings"))
+          .getRestorePath();
+    } catch (Throwable t) {
```

Comment:
We shouldn't be catching throwable here.

```suggestion
    } catch (FieldNotFoundException e) {
```